### PR TITLE
Add the integration testing for the bundles.

### PR DESCRIPTION
**Pull request by OleksiiZubko**
This is a working solution for testing generated and published bundles.
The mocha tesh framework is used for writing tests as it uses for writing unit tests.
Also adds the running tests for generated bundles to the PSV job.

Resolves: OLPEDGE-468, OLPEDGE-481
Relates-To: OLPEDGE-845

Signed-off-by: Oleksii Zubko <ext-oleksii.zubko@here.com>

### CODE REVIEW

File: tests/integration/bundles/umd/utils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    puppeteerCommand(arg: {
+      type: string;
+      args: {
+        [key: string]: string
+      }[]
+    }): Promise<any>;
+  }
+
+declare var window: ExtendedWindow;
+
```
Comment (OleksiiZubko) [inline tests/integration/bundles/umd/utils.ts]: @todo add comment

File: tests/integration/bundles/umd/olp-sdk-dataservice-api-testCases.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        it: "QueryApi should be defined",
+        callback: () => {
+            assert(QueryApi.getChangesById !== undefined);
+            assert(QueryApi.getPartitionsById !== undefined);
+            assert(QueryApi.quadTreeIndex !== undefined);
+            assert(QueryApi.quadTreeIndexVolatile !== undefined);
+        }
+    },
+    {
+        it: "VolatileBlobApi should be defined",
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply tests/integration/bundles/umd/olp-sdk-dataservice-api-testCases.ts]: Please change here and in all the similar occurrences in other files below.

File: tests/integration/bundles/umd/olp-sdk-generated.test.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { jsLoaderFactory } from "./utils";
+import { OlpSdkDataserviceApiTestCases } from "./olp-sdk-dataservice-api-testCases";
+import { OlpSdkAuthenticationTestCases } from "./olp-sdk-authentication-testCases";
+import { OlpSdkDataserviceReadTestCases } from "./olp-sdk-dataservice-read-testCases";
+
+describe("Test generated olp-edge-datastore-api", () => {
```
Comment (HalynaDumych) [inline tests/integration/bundles/umd/olp-sdk-generated.test.ts]: generated from?
Also, applies to the similar occurrences below.

File: tests/integration/bundles/umd/olp-sdk-published.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { jsLoaderFactory } from "./utils";
+import { OlpSdkDataserviceApiTestCases } from "./olp-sdk-dataservice-api-testCases";
+import { OlpSdkAuthenticationTestCases } from "./olp-sdk-authentication-testCases";
+import { OlpSdkDataserviceReadTestCases } from "./olp-sdk-dataservice-read-testCases";
+
+describe("Test published olp-edge-datastore-api", () => {
```
Comment (HalynaDumych) [inline tests/integration/bundles/umd/olp-sdk-published.test.ts]:
```suggestion
describe("Test published to olp-edge-datastore-api", () => {
```

File: tests/integration/bundles/umd/olp-sdk-published.test.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  before(jsLoaderFactory({url: "https://unpkg.com/@here/olp-sdk-dataservice-api/bundle.umd.min.js"}));
+
+  OlpSdkDataserviceApiTestCases.forEach(testCase => {
+    it(testCase.it, testCase.callback);
+  });
+});
+
+describe("Test published olp-edge-datastore-read", () => {
```
Comment (HalynaDumych) [inline tests/integration/bundles/umd/olp-sdk-published.test.ts]:
```suggestion
describe("Test published to olp-edge-datastore-read", () => {
```

File: tests/integration/bundles/umd/olp-sdk-published.test.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  before(jsLoaderFactory({url: "https://unpkg.com/@here/olp-sdk-dataservice-read/bundle.umd.min.js"}));
+
+  OlpSdkDataserviceReadTestCases.forEach(testCase => {
+    it(testCase.it, testCase.callback);
+  });
+});
+
+describe("Test published olp-sdk-authentication", () => {
```
Comment (HalynaDumych) [inline tests/integration/bundles/umd/olp-sdk-published.test.ts]:
```suggestion
describe("Test published to olp-sdk-authentication", () => {
```

**Pull request merged**